go/common/enclave.go

```diff
@@ -110,20 +110,13 @@ type Enclave interface {
 
 // BlockSubmissionResponse is the response sent from the enclave back to the node after ingesting a block
 type BlockSubmissionResponse struct {
-	BlockHeader             *types.Header             // The header of the consumed block.
 	UpdatedHeadRollup       bool                      // Whether the block contained a new head rollup for the canonical rollup chain.
 	IngestedRollupHeader    *Header                   // The header of the winning rollup contained in the ingested block, if any.
 	ProducedSecretResponses []*ProducedSecretResponse // if L1 block contained secret requests then there may be responses to publish
 	SubscribedLogs          map[rpc.ID][]byte         // The logs produced by the block and all its ancestors for each subscription ID.
 	RejectError             *BlockRejectError         // this is set if block was rejected, contains information about what block to submit next
 }
 
-// ProduceGenesisResponse is the response sent from the enclave back to the node after requesting the production of the genesis rollup.
-type ProduceGenesisResponse struct {
```

Comment:
As above. We don't need a special type any more, we can just return the genesis rollup.